### PR TITLE
rework pacbio store service

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
Closes https://github.com/Clinical-Genomics/improve-order-flow/issues/113
Rework the Pacbio store service so that it receives a PacbioOrder instead of an OrderIn

### Added

- Fixture for Pacbio order

### Changed

- Pacbio store service in the ordering
- Pacbio submitter

### Fixed

- Tests

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/orders/store_order_services/store_pacbio_order_service.py [hunk 1]
```diff
@@ -58,56 +61,82 @@ def order_to_status(order: OrderIn) -> dict:
...
+        The stored data objects are:
+        - Order
+        - Samples
+        - For each Sample, a Case
+        - For each Sample, a relationship between the Sample and its Case
+        """
+        # Add order
+        status_db_order: Order = self._create_db_order(order=order)
         new_samples = []
         with self.status_db.session.no_autoflush:
```
Comment (diitaz93) [inline cg/services/orders/store_order_services/store_pacbio_order_service.py:76]: consider wrapping this to decouple from sqlalchemy

Comment (diitaz93) [reply cg/services/orders/store_order_services/store_pacbio_order_service.py:76]: in all storing services

File: cg/services/orders/store_order_services/store_pacbio_order_service.py [hunk 2]
```diff
@@ -19,95 +23,96 @@ def __init__(self, status_db: Store, lims_service: OrderLimsService):
...
-        """Store PacBio samples and cases in the StatusDB."""
-        customer: Customer = self.status_db.get_customer_by_internal_id(
-            customer_internal_id=customer_id
-        )
-        status_db_order = Order(
-            customer=customer,
-            order_date=datetime.now(),
-            ticket_id=int(ticket_id),
-        )
+    def store_order_data_in_status_db(self, order: PacbioOrder) -> list[Sample]:
```
Comment (diitaz93) [inline cg/services/orders/store_order_services/store_pacbio_order_service.py:40]: maybe this should be a hidden method (`_`prefix)

File: tests/fixture_plugins/orders_fixtures/orders_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        user_id=0,
+        delivery_type=PacbioDeliveryType.BAM,
+        samples=samples,
+        name="PacbioOrder",
+    )
+    order._generated_ticket_id = int(ticket_id)
+    return order
+
+
+@pytest.fixture
```
Comment (diitaz93) [inline tests/fixture_plugins/orders_fixtures/orders_fixtures.py:63]: moved all the following fixtures from `tests/fixture_plugins/orders_fixtures/status_data_fixtures.py` which should be deprecated at the end of the store reworking

File: tests/services/orders/order_store_service/test_pacbio_order_service.py [hunk 4]
```diff
@@ -1,63 +1,42 @@
...
-    assert first_sample["priority"] == "priority"
-    assert first_sample["tumour"] is False
-    assert first_sample["volume"] == "25"
-
-    # THEN the other sample is a tumour
-    assert data["samples"][1]["tumour"] is True
+from tests.store_helpers import StoreHelpers
 
 
 def test_store_order(
```
Comment (diitaz93) [inline tests/services/orders/order_store_service/test_pacbio_order_service.py:13]: rename to the actual function that is being tested

File: tests/services/orders/order_store_service/test_pacbio_order_service.py [hunk 5]
```diff
@@ -1,63 +1,42 @@
...
     """Test that a PacBio order is stored in the database."""
     # GIVEN a basic store with no samples and a PacBio order
     assert not base_store._get_query(table=Sample).first()
     assert base_store._get_query(table=Case).count() == 0
 
+    # GIVEN that a Pacbio application exists in the store
+    helpers.ensure_application_version(
+        store=base_store,
+        application_tag="LWPBELB070",
+        prep_category=SeqLibraryPrepCategory.WHOLE_EXOME_SEQUENCING,
```
Comment (islean) [inline tests/services/orders/order_store_service/test_pacbio_order_service.py:28]:
```suggestion
        prep_category=SeqLibraryPrepCategory.WHOLE_GENOME_SEQUENCING,
```

File: cg/services/orders/store_order_services/store_pacbio_order_service.py [hunk 6]
```diff
@@ -19,95 +23,96 @@ def __init__(self, status_db: Store, lims_service: OrderLimsService):
...
+        """
+        Store all order data in the Status database for a Pacbio order. Return the samples.
+        The stored data objects are:
+        - Order
+        - Samples
+        - For each Sample, a Case
+        - For each Sample, a relationship between the Sample and its Case
+        """
+        # Add order
```
Comment (islean) [inline cg/services/orders/store_order_services/store_pacbio_order_service.py:49]:
```suggestion
```

File: cg/services/orders/store_order_services/store_pacbio_order_service.py (hunk 2 above)
Comment (islean) [reply cg/services/orders/store_order_services/store_pacbio_order_service.py:40]: Maybe. I don't think it bothers me too much though

**Review by islean — APPROVED** [review]
Looks great!

**Pull request merged**